### PR TITLE
fix : adds the arrayvec dependency

### LINKED ISSUE #2276: Use `ArrayVec` in `key::SortKey`
This would simplify the code and allow us to not initialize data past the key length improving performance of generating the compressed sort keys.

### PR DESCRIPTION
**Pull request by Harshit933**
This commit adds the arrayvec dependency to the sortKey.

Potential fix #2276

### CODE REVIEW

File: bitcoin/src/crypto/key.rs [hunk 1]
```diff
@@ -220,8 +217,8 @@ impl From<PublicKey> for XOnlyPublicKey {
 }
 
 /// An opaque return type for PublicKey::to_sort_key
-#[derive(Debug, Hash, PartialEq, Eq, PartialOrd, Ord, Clone, Copy)]
```
Comment (Kixunil) [inline bitcoin/src/crypto/key.rs:223]: `Copy` should stay, there's no reason to not have it. If `ArrayVec` doesn't support it we will have to figure out something else.

Comment (Harshit933) [reply bitcoin/src/crypto/key.rs:223]: Yes, `ArrayVec` does not support it for now. Should I close this PR then?

Comment (Kixunil) [reply bitcoin/src/crypto/key.rs:223]: So I investigated it and concluded the only solution is #2287.

If you rebase your PR on top of mine you should be able to simply remove `arrayvec` from `Cargo.toml` and use `internals::array_vec::ArrayVec` here and everything should work fine.

And for the future: the reason embedded failed is that you forgot to set `default-features = false` in `Cargo.toml`.

Comment (Harshit933) [reply bitcoin/src/crypto/key.rs:223]: Thanks for this. I'll just wait for it to get merged then I'll just rebase it on top of master

**Comment by Harshit933** [comment]
Updated the `arrayvec` according to #2287

**Review by Kixunil — APPROVED** [review]
ACK 8783d526bdf397d86763ce29f81fd2f9d8952c99

**Comment by tcharding** [comment]
Code changes look good, thanks for the contribution! The commit message is stale, can you update it please. If you are interested in improving your git skills check out this blog on writing good commit logs: https://cbea.ms/git-commit/

**Review by apoelstra — APPROVED** [review]
ACK 8783d526bdf397d86763ce29f81fd2f9d8952c99

**Comment by Harshit933** [comment]
> Code changes look good, thanks for the contribution! The commit message is stale, can you update it please. If you are interested in improving your git skills check out this blog on writing good commit logs: https://cbea.ms/git-commit/

Sure, I'll check it out and update the commit.

**Comment by Kixunil** [comment]
Too late for that now, but I think it's not a big deal here.

**Comment by apoelstra** [comment]
Sorry, I meant to hold off on merging this until I gave you a chance to fix it, but I forgot.

I should update my ACK script to allow writing "ACK <x> except..." which will catch my eye.

**Comment by Harshit933** [comment]
I'll take care of it from next time.

**Pull request merged**